### DISCUSSION TITLE
@property

### DISCUSSION THREAD

**Discussion by IranMine123** [tailwindlabs/tailwindcss#11986] [Ideas]
Hello, I just sent this link it explains the feature completely. please add this to tailwind. tysm
# CSS Custom Properties
https://developer.mozilla.org/en-US/docs/Web/CSS/@property
> The @property [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) [at-rule](https://developer.mozilla.org/en-US/docs/Web/CSS/At-rule) is part of the [CSS Houdini](https://developer.mozilla.org/en-US/docs/Web/Guide/Houdini) umbrella of APIs. It allows developers to explicitly define their [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*), allowing for property type checking and constraining, setting default values, and defining whether a custom property can inherit values or not.
The @property rule represents a custom property registration directly in a stylesheet without having to run any JS. Valid @property rules result in a registered custom property, as if [registerProperty()](https://developer.mozilla.org/en-US/docs/Web/API/CSS/registerProperty_static) had been called with equivalent parameters.

## Syntax
```
@property --property-name {
  syntax: "<color>";
  inherits: false;
  initial-value: #c0ffee;
}
```

**Comment by lukewarlow** [comment]
What would the API for this look like for you? It's a rather complicated syntax I can't see how Tailwind could add something ergonomic for this nor can I really think how it would be useful.

**Reply by cod3blox** [reply]
@property enables to interpolate custom properies. It is a really nice option.
Examples: [We can finally animate CSS gradient](https://dev.to/afif/we-can-finally-animate-css-gradient-kdk)

**Reply by jshmllr** [reply]
I just read about this this morning too! *bump

**Reply by lukewarlow** [reply]
It's still unclear to me how people would want this to work within tailwind?

**Reply by renet** [reply]
I know this is kind of an old issue. But is it still alive, or are there any updates on this? I imagine something like `hover:@custom-prop-red-500` to set a custom @property named `--custom-prop` (which has syntax `<color>`) to `var(--color-red-500)`. Or alternatively `hover:(--custom-prop)-red-500`, which will likely be a bit more misleading, as it can be confused with the custom property syntax meant for assigning values (`var(--custom-prop)`). Custom @properties would need to be defined in the tailwind config file, of course.

**Comment by alex-krasikau** [comment]
The @property doesn't belong to any element in the DOM, and it feels inappropriate to add it as a utility class. However, it is possible to use @property in the [CSS](https://tailwindcss.com/docs/adding-custom-styles#using-css-and-layer), which doesn't have any downsides or deviations from the Tailwind ideas.

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@property --property-name {
  syntax: "<color>";
  inherits: false;
  initial-value: #c0ffee;
}
```
